feat(libp2p): add `SwarmBuilder` (#4120)
paritytech/rust-libp2p: libp2p/CHANGELOG.md

> ## 0.52.4

Reviewer: ```suggestion
## 0.52.4 - unreleased
```

- Introduce the new `libp2p::SwarmBuilder`.

>   Users should use the new `libp2p::SwarmBuilder` instead of the now deprecated `libp2p::swarm::SwarmBuilder`.

Reviewer: ```suggestion
- Introduce a new `libp2p::SwarmBuilder` in favor of the now deprecated `libp2p::swarm::SwarmBuilder`.
```

Author: 80b8657f3

>   See `libp2p::SwarmBuilder` docs on how to use the new builder.

Reviewer: This is kind of implied, no?

Author: I want to particularly stress this instance, as the docs show the order on which to call the builder methods. Feel strongly about it @thomaseizinger?
